File: modules/spring-core/testdata/java/providers/linemarkers/beans/one_bean_one_dependency/TestConfiguration.java
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.stereotype.Component;

class FooAOneBeanOneDependency {}

class FooBOneBeanOneDependency {}

@Component
class FooCOneBeanOneDependency {}

@Configuration
public class TestConfiguration {
    @Bean
    FooAOneBeanOneDependency fooAOneBeanOneDependency() { return new FooAOneBeanOneDependency(); }
}

@Component
class FooComponentOneBeanOneDependency {
    @Autowired
    FooAOneBeanOneDependency fooAOneBeanOneDependency;

    @Autowired
    FooBOneBeanOneDependency fooBOneBeanOneDependency;

    @Autowired
    FooCOneBeanOneDependency fooCOneBeanOneDependency;
}

File: modules/spring-core/testdata/kotlin/providers/linemarkers/beans/one_bean_one_dependency/TestConfiguration.kt
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.stereotype.Component

class FooAOneBeanOneDependency

internal class FooBOneBeanOneDependency

@Component
internal class FooCOneBeanOneDependency

@Configuration
open class TestConfiguration {
    @Bean
    open fun fooAOneBeanOneDependency(): FooAOneBeanOneDependency {
        return FooAOneBeanOneDependency()
    }
}

@Component
internal class FooComponentOneBeanOneDependency {
    @Autowired
    var fooAOneBeanOneDependency: FooAOneBeanOneDependency? = null

    @Autowired
    var fooBOneBeanOneDependency: FooBOneBeanOneDependency? = null

    @Autowired
    var fooCOneBeanOneDependency: FooCOneBeanOneDependency? = null
}